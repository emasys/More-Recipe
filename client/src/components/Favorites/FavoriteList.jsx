import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Fade from 'react-reveal/Fade';

const FavoriteList = ({ favorites }) => {
  console.log(favorites);
  if (favorites.userFav) {
    if (favorites.userFav.favorites.length < 1) {
      return (
        <div className="text-center error-message">
          <div>
            <img src="../img/logo.png" alt="logo" />
            <h4 className="p-3 m-2">You don't have any favorite recipe yet</h4>
            <p className="p-3 m-2">
              <Link to="/catalog" className="btn btn-outline-dark">
                check out amazing recipes
              </Link>
            </p>
          </div>
        </div>
      );
    }
    return favorites.userFav.favorites.map((item, index) => (
      <div
        key={index}
        className="col-lg-3 col-sm-10 mb-20  col-md-4 animate-catalog"
        data-animate="bounceIn"
        data-duration="1.0s"
        data-delay="0.1s"
        data-offset="100"
      >
        <div>
          <Fade bottom>
            <Link to={`/recipe/${item.recipeId}`} className=" hvr-grow-shadow">
              <div className={`card animate`}>
                <img
                  className="card-img-top img-box"
                  src={item.Recipe.foodImg}
                  alt="Card image cap"
                />
                <div className="card-body p-0 text-center social-icons">
                  <span className="tag bg-danger">{item.Recipe.category}</span>
                  <h4 className="card-title custom-bg bg-dark p-2 m-0 text-truncate ">
                    {item.Recipe.name.length > 25 ?
                      item.Recipe.name.slice(0, 15).concat('...') :
                      item.Recipe.name}
                  </h4>
                  <div className="card-body p-5 text-left bg-light text-dark">
                    <p className="crop-text">{item.Recipe.description}</p>
                  </div>
                  <span>
                    <i className="fa fa-heart-o" aria-hidden="true" />
                    {item.Recipe.favorite}
                  </span>
                  <span>
                    <i className="fa fa-thumbs-o-up" aria-hidden="true" />
                    {item.Recipe.upvote}
                  </span>
                  <span>
                    <i className="fa fa-thumbs-o-down" aria-hidden="true" />
                    {item.Recipe.downvote}
                  </span>
                  <span>
                    <i className="fa fa-eye" aria-hidden="true" />
                    {item.Recipe.views}
                  </span>
                  <span>
                    <i className="fa fa-comment-o" aria-hidden="true" />
                    {item.Recipe.comments}
                  </span>
                </div>
              </div>
            </Link>
          </Fade>
        </div>
      </div>
    ));
  }
  return null;
};

FavoriteList.propTypes = {
  favorites: PropTypes.object
};

export default FavoriteList;
