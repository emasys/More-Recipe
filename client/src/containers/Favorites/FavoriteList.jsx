import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import approx from 'approximate-number';
import { uniq } from 'lodash';

/**
 *
 *
 * @param {object} { favorites }
 * @returns {JSX.Element} react element
 */
const FavoriteList = ({ favorites }) => {
  if (favorites) {
    if (favorites.length < 1) {
      return (
        <div className="text-center error-message">
          <div>
            <img src="../img/logo.png" alt="logo" />
            <h4 className="p-3 m-2">You do not have any favorite recipe yet</h4>
            <p className="p-3 m-2">
              <Link to="/catalog" className="btn btn-outline-dark">
                check out amazing recipes
              </Link>
            </p>
          </div>
        </div>
      );
    }
    const uniqueFavorites = uniq(favorites);
    return uniqueFavorites.map(item => (
      <div key={item.recipeId} className="row">
        <div
          data-aos="fade-up"
          data-aos-duration="2000"
          className="col-lg-12 col-sm-12 mb-20 mt-50 col-md-12"
        >
          <Link to={`/recipe/${item.recipeId}`} className="hvr-push">
            <div className="card animate">
              <img
                className="card-img-top img-box"
                src={item.Recipe.foodImg}
                alt="itemImage"
              />
              <div className="card-body p-0 text-center social-icons">
                <span className="tag bg-danger">{item.Recipe.category}</span>
                <h4 className="card-title custom-bg bg-dark p-2 m-0 text-truncate ">
                  {item.Recipe.name.length > 25 ?
                    item.Recipe.name.slice(0, 25).concat('...') :
                    item.Recipe.name}
                </h4>
                <div className="card-body p-5 text-left bg-wheat text-dark">
                  <p className="crop-text">{item.Recipe.description}</p>
                </div>
                <span>
                  <i className="fa fa-heart-o" aria-hidden="true" />
                  {approx(item.Recipe.favorite)}
                </span>
                <span>
                  <i className="fa fa-thumbs-o-up" aria-hidden="true" />
                  {approx(item.Recipe.upvote)}
                </span>
                <span>
                  <i className="fa fa-thumbs-o-down" aria-hidden="true" />
                  {approx(item.Recipe.downvote)}
                </span>
                <span>
                  <i className="fa fa-eye" aria-hidden="true" />
                  {approx(item.Recipe.views)}
                </span>
                <span>
                  <i className="fa fa-comment-o" aria-hidden="true" />
                  {approx(item.Recipe.comments)}
                </span>
              </div>
            </div>
          </Link>
        </div>
      </div>
    ));
  }
  return null;
};

FavoriteList.propTypes = {
  favorites: PropTypes.array.isRequired
};

export default FavoriteList;
