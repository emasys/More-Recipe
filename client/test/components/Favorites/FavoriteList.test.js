import React from 'react';
import { shallow, mount } from 'enzyme';
import FavoriteList from '../../../src/components/Favorites/FavoriteList';

const propsEmpty = {
  userFav: {
    favorites: []
  }
};

const favorites = {
  userFav: {
    favorites: [
      {
        id: 44,
        recipeId: 27,
        userId: 5,
        createdAt: '2018-01-10T04:33:00.481Z',
        updatedAt: '2018-01-10T04:33:00.481Z',
        Recipe: {
          id: 27,
          name: 'How to cook something cool',
          ingredients: [
            'water',
            'salt',
            'oil',
            'preloading is working fine now',
            'new stuff'
          ],
          direction: 'just do it, really, really?',
          description: 'just do it, really?',
          category: 'Appetisers',
          foodImg:
            'https://res.cloudinary.com/emasys/image/upload/v1515914279/ruwl3dswlgju81udttvd.jpg',
          userId: 5,
          upvote: 1,
          downvote: 0,
          reactionUp: [5],
          reactionDown: [],
          views: 1,
          comments: 38,
          favorite: 1,
          createdAt: '2018-01-09T19:43:09.761Z',
          updatedAt: '2018-01-14T07:18:00.149Z'
        }
      }
    ]
  }
};

test('should render favorite list correctly with no favorite recipe', () => {
  const wrapper = shallow(<FavoriteList favorites={propsEmpty} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render favorite list correctly with one favorite recipe', () => {
  const wrapper = mount(<FavoriteList favorites={favorites} />);
  expect(wrapper).toMatchSnapshot();
});
