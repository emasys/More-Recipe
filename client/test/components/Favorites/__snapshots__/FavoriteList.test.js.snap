// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render favorite list correctly with no favorite recipe 1`] = `
<div
  className="text-center error-message"
>
  <div>
    <img
      alt="logo"
      src="../img/logo.png"
    />
    <h4
      className="p-3 m-2"
    >
      You don't have any favorite recipe yet
    </h4>
    <p
      className="p-3 m-2"
    >
      <Link
        className="btn btn-outline-dark"
        replace={false}
        to="/catalog"
      >
        check out amazing recipes
      </Link>
    </p>
  </div>
</div>
`;

exports[`should render favorite list correctly with one favorite recipe 1`] = `
Array [
  <div
    className="col-lg-3 col-sm-10 mb-20  col-md-4 animate-catalog"
    data-animate="bounceIn"
    data-delay="0.1s"
    data-duration="1.0s"
    data-offset="100"
    key="0"
  />,
]
`;
